vm/src/obj/objdict.rs

```diff
@@ -228,6 +228,13 @@ impl PyDictRef {
         PyDictRef::merge(&self.entries, dict_obj, kwargs, vm)
     }
 
+    fn pop(self, key: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        match self.entries.borrow_mut().pop(vm, &key) {
```

Comment:
You may want to use `map_err` here. Furthermore, I think we do not require the map the result. Can't we do without this match statement and return whatever the `pop` function on entries returns?